samcli/commands/build/build_context.py

```diff
@@ -170,7 +170,11 @@ def resources_to_build(self):
         return result
 
     @property
-    def is_specific_resource(self):
+    def is_building_specific_resource(self):
+        """
+        Checks whether customer requested to build specific resource
```

Comment:
customer requested to build a specific resource alone in isolation, by specifying function_identifier to the build command. 

Something like that makes it more clear.